include/ic_components.h

```diff
@@ -0,0 +1,124 @@
+#pragma once
+
+#include "ic_graphics.h"
+
+#include <glm/glm.hpp>
+#include <imgui.h>
+
+#include <vector>
+
+namespace IC {
+    class GameObjectComponent {
+    public:
+        GameObjectComponent();
+        ~GameObjectComponent();
```

Comment:
Mark this `virtual`. Rule of thumb is all virtual classes should have its dtor also virtual.